src/clue.js

```diff
@@ -2,25 +2,157 @@
+        image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        color: "blue",
+    },
+    mrMustardv = {
+        firstName: "Jack",
+        lastName: "Mustard",
+        occupation: "Retired Football player",
+        age: 62,
+        description: "He is a former football player who tries to get by on his former glory",
+        image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+        color: "yellow",
+    }
+];
+
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+        {name: "Dining Room"},
+        {name: "Conservatory"},
+        {name: "Kitchen"},
+        {name: "Study"},
+        {name: "Library"},
+        {name: "Billiard Room"},
+        {name: "Lounge"},
+        {name: "Ballroom"},
+        {name: "Hall"},
+        {name: "Spa"},
+        {name: "Living Room"},
+        {name: "Observatory"},
+        {name: "Theater"},
+        {name: "Guest House"},
+        {name: "Patio"},
+    
+];
+
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+                    {name: "rope",
+                    weight: 10,
+                    },
+                    { 
+                    name: "knife",
+                    weight: 8, 
+                    },
+                    {
+                    name: "candlestick",
+                    weight: 20, 
+                    },
+                    {
+                    name: "dumbbell", 
+                    weight: 30, 
+                    },
+                    {
+                    name: "poison", 
+                    weight: 2, 
+                    },
+                    {
+                    name: "axe", 
+                    weight: 15, 
+                    },
+                    {
+                    name: "bat", 
+                    weight: 13,
+                    },
+                    {
+                    name: "trophy", 
+                    weight: 25,
+                    },
+                    {
+                    name: "pistol",
+                    weight: 20,
+                    },
+];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(arr) {
+    let randomCard = Math.floor(Math.random() * arr.length);
+
+    if (arr.length === 0) {
+        return undefined;
+    } else if (arr.length === 1) {
+        return arr[0];
+    } else {
+        return arr[randomCard];
+    }
+
+}
+
+function pickMystery() {
+    let suspect = selectRandom(suspectsArray);
+    let weapon = selectRandom(weaponsArray);
+    let room = selectRandom(roomsArray);
+
+    const mystery = {suspect, weapon, room};
+
+    return mystery;
 
-function pickMystery() {}
+}
 
 
 // ITERATION 3
 
-function revealMystery() {}
+function revealMystery(envelope) {
+    return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the ${envelope.weapon.name} in the ${envelope.room.name}!`
+}
```

Comment:
Here, you have everything correct, however, it is good practice if you destructure your `envelope` argument.
By destructuring, it clearly expresses which properties you're using within the function.
By extracting the properties into individual variables (suspect, weapon, and room), you can directly reference them in your return statement without needing to use `envelope.suspect`, `envelope.weapon`, and `envelope.room` every time. This simplifies the code and improves readability.
It would look something like this:

```
function revealMystery(envelope) {
  let { suspect, weapon, room } = envelope;

  return `${suspect.firstName} ${suspect.lastName} killed Mr. Boddy using the ${weapon.name} in the ${room.name}!`;
}
```